orc/src/main/java/org/apache/iceberg/orc/avro/GenericDataOrcWriter.java

```diff
@@ -0,0 +1,526 @@
+  }
+
+  static class UUIDConverter implements Converter<UUID> {
+    private final BytesConverter bytesConverter;
+
+    UUIDConverter() {
+      bytesConverter = new BytesConverter();
+    }
+
+    @Override
+    public Class<UUID> getJavaClass() {
+      return UUID.class;
+    }
+
+    private static byte[] asBytes(UUID uuid) {
+      ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+      bb.putLong(uuid.getMostSignificantBits());
+      bb.putLong(uuid.getLeastSignificantBits());
+      return bb.array();
+    }
+
+    @Override
+    public void addValue(int rowId, UUID data, ColumnVector output) {
+      bytesConverter.addValue(rowId, data != null ? asBytes(data) : null, output);
+    }
+  }
+
+  static class FixedConverter implements Converter<GenericData.Fixed> {
+    private final BytesConverter bytesConverter;
+
+    FixedConverter() {
+      bytesConverter = new BytesConverter();
+    }
+
+    @Override
+    public Class<GenericData.Fixed> getJavaClass() {
+      return GenericData.Fixed.class;
+    }
+
+    @Override
+    public void addValue(int rowId, GenericData.Fixed data, ColumnVector output) {
+      bytesConverter.addValue(rowId, data != null ? data.bytes() : null, output);
+    }
+  }
+
+  static class BytesConverter implements Converter<byte[]> {
+    @Override
+    public Class<byte[]> getJavaClass() {
+      return byte[].class;
+    }
+
+    public void addValue(int rowId, byte[] data, ColumnVector output) {
+      if (data == null) {
+        output.noNulls = false;
+        output.isNull[rowId] = true;
+      } else {
+        output.isNull[rowId] = false;
+        // getBinary always makes a copy, so we don't need to worry about it
+        // being changed behind our back.
+        ((BytesColumnVector) output).setRef(rowId, data, 0, data.length);
+      }
+    }
+  }
+
+  static class TimestampConverter implements Converter<Long> {
+    @Override
+    public Class<Long> getJavaClass() {
+      return Long.class;
+    }
+
+    public void addValue(int rowId, Long data, ColumnVector output) {
+      if (data == null) {
+        output.noNulls = false;
+        output.isNull[rowId] = true;
+      } else {
+        output.isNull[rowId] = false;
+        TimestampColumnVector cv = (TimestampColumnVector) output;
+        long micros = data;
+        cv.time[rowId] = micros / 1_000; // millis
+        cv.nanos[rowId] = (int) (micros % 1_000_000) * 1_000; // nanos
+      }
+    }
+  }
+
+  static class Decimal18Converter implements Converter<BigDecimal> {
+    private final int precision;
+    private final int scale;
+
+    Decimal18Converter(TypeDescription schema) {
+      this.precision = schema.getPrecision();
+      this.scale = schema.getScale();
+    }
+
+    @Override
+    public Class<BigDecimal> getJavaClass() {
+      return BigDecimal.class;
+    }
+
+    public void addValue(int rowId, BigDecimal data, ColumnVector output) {
+      // TODO: validate precision and scale from schema
+      if (data == null) {
```

Comment:
maybe we abstract out the common pieces here, like we do in the reader side. https://github.com/apache/incubator-iceberg/blob/master/orc/src/main/java/org/apache/iceberg/orc/OrcValueReader.java [Note this is slightly changed, but we did had a Converter on the read side too with with common bits abstracted]

Reply:
This class is only used for a test. I think we should convert the test over to using Iceberg generics instead of Avro generics. I'll work on this later today.

Reply:
Yeah, I created this just for the tests. I'd be great to use Iceberg. Thanks @rdblue.

Reply:
I just opened #1070 with the refactor.